docs/source/serving.rst

```diff
@@ -1,12 +1,468 @@
 (Part 3) Serving on vLLM, SGLang, ExecuTorch
-------------------------------------------------
+============================================
 
-TorchAO provides an end-to-end pre-training, fine-tuning, and serving
-model optimization flow by leveraging our quantization and sparsity
-techniques integrated into our partner frameworks. This is part 3 of 3
-such tutorials showcasing this end-to-end flow, focusing on the
-serving step.
+TorchAO provides an end-to-end pre-training, fine-tuning, and serving model optimization flow by leveraging our quantization and sparsity techniques integrated into our partner frameworks. This is part 3 of 3 such tutorials showcasing this end-to-end flow, focusing on the serving step.
 
 .. image:: ../static/e2e_flow_part3.png
 
-(Coming soon!)
+This tutorial demonstrates how to perform post-training quantization and deploy models for inference using torchao as the underlying optimization engine, seamlessly integrated through HuggingFace Transformers, vLLM, and ExecuTorch.
+
+.. contents::
+   :local:
+   :depth: 2
+
+Post-training Quantization with HuggingFace
+-------------------------------------------
+
+HuggingFace Transformers provides seamless integration with torchao quantization. The ``TorchAoConfig`` automatically applies torchao's optimized quantization algorithms during model loading. For this example, we'll use `Float8DynamicActivationFloat8WeightConfig` on the Phi-4 mini-instruct model.
+
+.. code-block:: bash
+    pip install git+https://github.com/huggingface/transformers@main
+    pip install --pre torchao --index-url https://download.pytorch.org/whl/nightly/cu126
+    pip install torch
+    pip install accelerate
+
+.. code-block:: python
+
+    import torch
+    from transformers import AutoModelForCausalLM, AutoTokenizer, TorchAoConfig
+
+    model_id = "microsoft/Phi-4-mini-instruct"
+
+    from torchao.quantization import Float8DynamicActivationFloat8WeightConfig, PerRow
+    quant_config = Float8DynamicActivationFloat8WeightConfig(granularity=PerRow())
+    quantization_config = TorchAoConfig(quant_type=quant_config)
+    quantized_model = AutoModelForCausalLM.from_pretrained(model_id, device_map="auto", torch_dtype=torch.bfloat16, quantization_config=quantization_config)
+    tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+    # Push to hub
+    USER_ID = "YOUR_USER_ID"
+    MODEL_NAME = model_id.split("/")[-1]
+    save_to = f"{USER_ID}/{MODEL_NAME}-float8dq"
+    quantized_model.push_to_hub(save_to, safe_serialization=False)
+    tokenizer.push_to_hub(save_to)
+
+    # Manual Testing
```

Comment:
I would split this into a separate code block and add some text in between, since everything below this line is technically not part of the user flow